### PR TITLE
NSDictionaryController branch

### PR DESCRIPTION
**Pull request by gcasa**

### CODE REVIEW

**Comment by gcasa** [comment]
> Why not NSExcludedKeysBinding?

The dangers of coding while tired.  I fixed the spelling.

**Comment by gcasa** [comment]
I have an example that tests bindings.  It uses NSTableView.  One is NSArrayController_test and the other is NSDictionaryController_test.  NSArrayController_test works on macOS10.15 but doesn't function on GNUstep at all.  I believe the issue is in NSTableView.

**Comment by fredkiefer** [comment]
> I have an example that tests bindings. It uses NSTableView. One is NSArrayController_test and the other is NSDictionaryController_test. NSArrayController_test works on macOS10.15 but doesn't function on GNUstep at all. I believe the issue is in NSTableView.

"doesn't function" is not a very helpful description. What happens and what should be happening instead? There are quite a few missing features in NSTableView and as for NSArrayController I only got that working with a horrible hack, I wrote about that last winter. You are welcome to improve on both classes.

**Comment by gcasa** [comment]
> > I have an example that tests bindings. It uses NSTableView. One is NSArrayController_test and the other is NSDictionaryController_test. NSArrayController_test works on macOS10.15 but doesn't function on GNUstep at all. I believe the issue is in NSTableView.
> 
> "doesn't function" is not a very helpful description. What happens and what should be happening instead? There are quite a few missing features in NSTableView and as for NSArrayController I only got that working with a horrible hack, I wrote about that last winter. You are welcome to improve on both classes.

I will work on both of them.   I saw your comment about the hack.  I wrote a bug on this:
https://github.com/gnustep/libs-gui/issues/87

File: Headers/AppKit/NSKeyValueBinding.h [hunk 1]
```diff
@@ -86,10 +86,16 @@
 
 @end
 
-// Keys in options dictionary
+// typedefs
+typedef NSString* NSBindingName;
```
Comment (fredkiefer) [inline Headers/AppKit/NSKeyValueBinding.h:90]: Now you define this type twice, it is already on line 95. BTW, the placement of the star is correct on line 95 and incorrect here.

File: Source/externs.m [hunk 2]
```diff
@@ -754,14 +754,21 @@
 NSString *NSAlignmentBinding = @"alignment";
```
Comment (fredkiefer) [inline Source/externs.m:754]: Shouldn't we change the types here as well?

**Comment by gcasa** [comment]
This PR uses a branch that has been superseded.

**Pull request closed without merging**